squape/object_tree.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import math
+
+import object
+
+import squish
+
+
+def children(obj, selector: dict = None) -> tuple:
+    """Filter the direct children of the specified object by the specified selector
+
+    Args:
+        obj (Squish object): A parent node.
+        selector (dict, optional): The selector is a dictionary of properties,
+        that must match for objects to be included into the result.
+        Defaults to {}, which means all objects pass the verification.
+
+    Returns:
+        tuple: the result of search among the object tree.
+
+    Examples:
+        children(object, {'type' : Button, 'visible' : True})
+    """
+    if selector is None:
+        selector = {}
+    children = object.children(obj)
+    return tuple(filter(lambda x: _filter_by_selector(x, selector), children))
+
+
+def find(obj, selector: dict = None, max_depth=None, _depth=0) -> tuple:
```

Comment:
1. same comments to naming, parameter type, and docstrings as previously
2. I think you don't need the `_depth` parameter at all, instead you could decrease max_depth for every recursive call.